### PR TITLE
docs(array): add docs about array function

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+title: 'Array Functions'
+---
+
+Array Functions
+
+## Syntax
+
+| Function    | Description             |  Example | Result
+| ----------- | ------------------ |  ---------- | --------|
+| `get(array1, index)`  | Get item from an array by index     |  `get([1,2], 2)`         |  2
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:11]: `get(array, index)`

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+| `length(array1)`  | Output the length of an array    |  `length([1,2])`         |  2
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:12]: `length(array)`

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+| `array_concat(array1, array2)`  | Concat two arrays     |  `array_concat([1,2], [3, 4]`         |  [1,2,3,4]
+| `array_contains(array, item)`  | Check if array has some item     |  `array_contains([1,2], 1)`         |  1
+| `array_indexof(array, item)`  | Returns the index of the element if the list contains the element.	list_contains([1, 2, NULL],      |  `array_indexof([1,2,9], 9);`         |  3
+| `array_slice(array, start, end)`  | Extract a sublist using slice conventions     |  `array_slice([1,21,32,4], 2, 3)`         |  `[21, 32]`
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:16]: `array_slice` function also support two arguments to extract sublist from `start` to the end.

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+| `array_sort(array)`  | Sort an array     |  `array_sort([1,4,3,2])`         |  `[1,2,3,4]`
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:17]: `array_sort` function have two additional optional arguments.
The second optional argument is the sort order, `ASC` and `DESC` are supported, the default value is `ASC`.
The third optional argument is the position of the `NULL` value, `NULLS FIRST` and `NULLS LAST` are supported, the default is `NULLS FIRST`.

File: docs/doc/15-sql-functions/04-array-functions/_category_.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (BohuTANG) [inline docs/doc/15-sql-functions/04-array-functions/_category_.json:1]:
```suggestion
{
    "label": "Array Functions"
 }
```

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+| `array_<aggr>(array)`  | Aggregate the array using aggr name(currently support: sum, count, avg, min, max, any)     |  `array_sum([1,2,3,4]`         |  10
+| `array_unique(array)`  | Counts the unique elements of an array      |  `array_unique([1,2,3,3,4])`         |  4
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:19]: Counts the number of unique elements in array other than NULL.

File: docs/doc/15-sql-functions/04-array-functions/index.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+title: 'Array Functions'
+---
+
+Array Functions
```
Comment (b41sh) [inline docs/doc/15-sql-functions/04-array-functions/index.md:5]: The index of array in array functions are 1-based.

File: docs/doc/15-sql-functions/04-array-functions/index.md (hunk 1 above)
Comment (BohuTANG) [reply docs/doc/15-sql-functions/04-array-functions/index.md:11]: image
Why databend index start from 1 not 0?

**Comment by BohuTANG** [comment]
ChatGPT with prompts: `think you are a snowflake expert, please improve this doc: <this-doc-markdown>`

image

File: docs/doc/15-sql-functions/04-array-functions/index.md (hunk 7 above)
Comment (BohuTANG) [reply docs/doc/15-sql-functions/04-array-functions/index.md:5]: Why 1-based for index?

Comment (sundy-li) [reply docs/doc/15-sql-functions/04-array-functions/index.md:5]: It's  SQL-99 standard. Snowflake is not following this.

Comment (BohuTANG) [reply docs/doc/15-sql-functions/04-array-functions/index.md:5]: This is an anti-user experience(I know ClickHouse did it), the index should be 0-based for an array.
I think snow does it right:
https://docs.snowflake.com/en/sql-reference/functions/get.html

File: docs/doc/15-sql-functions/04-array-functions/index.md (hunk 1 above)
Comment (sundy-li) [reply docs/doc/15-sql-functions/04-array-functions/index.md]: Almost `> 90%` of databases use 1 based index in SQL as SQL-99 standard.

Comment (BohuTANG) [reply docs/doc/15-sql-functions/04-array-functions/index.md]: Right, but not a good experience(Not many people are familiar with SQL-99 standard).
We should explicitly `(1-based)` it like duckdb:
https://duckdb.org/docs/sql/functions/nested#list-functions

**Comment by b41sh** [comment]
> ChatGPT with prompts: `think you are a snowflake expert, please improve this doc: <this-doc-markdown>`
> 
> image
> 

the index of array data type in Snowflake is 0-based

File: docs/doc/15-sql-functions/04-array-functions/index.md (hunk 7 above)
Comment (sundy-li) [reply docs/doc/15-sql-functions/04-array-functions/index.md:5]: Docs update.

Clickhouse is also 1 based, SQL is always 1-based, JDBC driver is 1 based.

**Review by b41sh — APPROVED** [review]

**Pull request merged**